### PR TITLE
Nextmillennium: add extra_info support

### PR DESCRIPTION
**Pull request by JacobKlein26**
Added the ability for users to pass custom flags using extra_info,

### CODE REVIEW

File: adapters/nextmillennium/nextmillenniumtest/exemplary/banner-empty-group-id.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
         "body":{
           "id": "testid",
           "ext": {
+            "nextMillennium": {},
```
Comment (SyntaxNode) [inline adapters/nextmillennium/nextmillenniumtest/exemplary/banner-empty-group-id.json:36]: Is it desired to send an empty object if the list is empty? If you add an omitempty flag to the json attribute you could skip creating an empty object (unless you need it on your side). ie:

```
NextMillennium nmExtNMM    `json:"nextMillennium"`
```
to
```
NextMillennium nmExtNMM    `json:"nextMillennium,omitempty"`
```

**Review by SyntaxNode — COMMENTED** [review]
Looking good. Please add unit test coverage for passing in extraInfo and verifying it's included in the request to your servers. Something similar to `huaweiads_test.go` would be great.

File: adapters/nextmillennium/nextmillenniumtest/exemplary/banner-empty-group-id.json (hunk 1 above)
Comment (JacobKlein26) [reply adapters/nextmillennium/nextmillenniumtest/exemplary/banner-empty-group-id.json:36]: I added a test case for extra_info,
When I add `omitempty` it doesn't actually change anything... I always believed `omitempty` only works on keys (meaning not objects) (or if it's a pointer...),
Either way, it doesn't bother us, so if it's ok with you, I would just leave it...

Comment (onkarvhanumante) [reply adapters/nextmillennium/nextmillenniumtest/exemplary/banner-empty-group-id.json:36]: @JacobKlein26 

`omitempty` was added on `NmmFlags`. But recommendation was to add omitempty on `nextMillennium` 


```
type nmExtNMM struct {
- 	NmmFlags []string `json:"nmmFlags,omitempty"`
+ 	NmmFlags []string `json:"nmmFlags"`
}
type nextMillJsonExt struct {
	Prebid nmExtPrebid `json:"prebid"`
	Prebid         nmExtPrebid `json:"prebid"`
-	NextMillennium nmExtNMM    `json:"nextMillennium"`
+ 	NextMillennium nmExtNMM    `json:"nextMillennium,omitempty"`
```

Comment (JacobKlein26) [reply adapters/nextmillennium/nextmillenniumtest/exemplary/banner-empty-group-id.json:36]: Hi @onkarvhanumante.
Thanks for clarifying.
I pushed another commit to add it. However, as far as I'm aware, adding omitempty on a type of struct makes no difference (the object itself will be empty but the nextMillennium key will still exist as an empty object).
The only way (from a quick google search) to remove it is to convert it to a pointer.
It is not important for us to remove it, but if for whatever reason you think I should do it, let me know and I'll do it.

Thank you!

**Review by onkarvhanumante — APPROVED** [review]

**Review by gargcreation1992 — APPROVED** [review]

**Pull request merged**